src/Setup.php

```diff
@@ -331,32 +332,80 @@ public function bcgov_block_theme_custom_settings() {
 		register_setting( 'bcgov-block-theme-settings-group', 'header_effect' );
 		register_setting( 'bcgov-block-theme-settings-group', 'enable_all_styles' );
 		register_setting( 'bcgov-block-theme-settings-group', 'custom_body_class' );
+		register_setting( 'bcgov-block-theme-settings-group', 'custom_csp_defaultsrc' );
+		register_setting( 'bcgov-block-theme-settings-group', 'custom_csp_scriptsrc' );
+		register_setting( 'bcgov-block-theme-settings-group', 'custom_csp_stylesrc' );
+		register_setting( 'bcgov-block-theme-settings-group', 'custom_csp_connectsrc' );
+		register_setting( 'bcgov-block-theme-settings-group', 'custom_csp_fontsrc' );
+		register_setting( 'bcgov-block-theme-settings-group', 'custom_csp_imgsrc' );
+		register_setting( 'bcgov-block-theme-settings-group', 'custom_csp_mediasrc' );
+		register_setting( 'bcgov-block-theme-settings-group', 'custom_csp_framesrc' );
 
 		$active_site_pattern_styles = esc_attr( get_option( 'active_site_pattern_styles' ) );
 		$header_effect              = esc_attr( get_option( 'header_effect' ) );
 		$enable_all_styles          = esc_attr( get_option( 'enable_all_styles' ) );
 		$custom_body_class          = esc_attr( get_option( 'custom_body_class' ) );
 
+		$custom_csp_defaultsrc = esc_attr( get_option( 'custom_csp_defaultsrc' ) );
+		$custom_csp_scriptsrc  = esc_attr( get_option( 'custom_csp_scriptsrc' ) );
+		$custom_csp_stylesrc   = esc_attr( get_option( 'custom_csp_stylesrc' ) );
+		$custom_csp_connectsrc = esc_attr( get_option( 'custom_csp_connectsrc' ) );
+		$custom_csp_fontsrc    = esc_attr( get_option( 'custom_csp_fontsrc' ) );
+		$custom_csp_imgsrc     = esc_attr( get_option( 'custom_csp_imgsrc' ) );
+		$custom_csp_mediasrc   = esc_attr( get_option( 'custom_csp_mediasrc' ) );
+		$custom_csp_framesrc   = esc_attr( get_option( 'custom_csp_framesrc' ) );
+
 		/*
 		* Initial settings to maintain configuration of CleanBC in production
 		* these will change to default BCGov configuration after v1.0.3.
 		*/
 		if ( empty( $active_site_pattern_styles ) ) {
-			add_option( 'enable_all_styles', 'bcgov' );
+			update_option( 'enable_all_styles', 'bcgov' );
 		}
 
 		if ( empty( $header_effect ) ) {
-			add_option( 'header_effect', 'hides' ); // will be: fixed.
+			update_option( 'header_effect', 'hides' ); // will be: fixed.
 		}
 
 		if ( empty( $enable_all_styles ) ) {
-			add_option( 'enable_all_styles', 'true' ); // will be: false.
+			update_option( 'enable_all_styles', 'false' ); // will be: false.
 		}
 
 		if ( empty( $custom_body_class ) ) {
-			add_option( 'custom_body_class', 'default' );
+			update_option( 'custom_body_class', 'default' );
+		}
+
+		if ( empty( $custom_csp_defaultsrc ) ) {
```

Comment:
My thought would be that these csp options would only have the overrides, and not the entire value.
My fear is they get overwritten, and things either start failing or create a security risk.
 Would it be better just to have it in the hook for the headers when you get_option to have the default there, that way when you are looking at bcgov_block_theme_wp_headers callback all the defaults are visible.